src/app/backend/deploy.go

```diff
@@ -224,3 +237,28 @@ func getLabelsMap(labels []Label) map[string]string {
 
 	return result
 }
+
+// Deploys an app based on the given yaml or json file.
+func DeployAppFromFile(spec *AppDeploymentFromFileSpec, client *client.Client) error {
+	// Creates temp file to use it in kubectl command
+	file, err := ioutil.TempFile("", spec.Name)
+	if err != nil {
+		return err
+	}
+	_,err = file.WriteString(spec.Content)
```

Comment:
spec.content can be byte[] instead of string. In case it is needed, frontend should be modified as well.